File: java/src/main/java/io/metaparticle/sync/examples/ElectionMain.java
package io.metaparticle.sync.examples;

import io.metaparticle.sync.Election;

import java.util.Random;

public class ElectionMain {
    public static void main(String[] args) throws InterruptedException {
        Random r = new Random();
        while (true) {
            final Object block = new Object();
            Election e = new Election(
                "test", args[0],
                () -> {
                    System.out.println("I am the master.");
                    synchronized(block) {
                        try {
                            block.wait();
                        } catch (InterruptedException ex) {
                            ex.printStackTrace();
                        }
                    }
                },
                () -> {
                    System.out.println("I lost the master.");
                    synchronized(block) {
                        block.notify();
                    }
                });
            new Thread(e).start();
            Thread.sleep((r.nextInt(15) + 25) * 1000);
            e.shutdown();
            Thread.sleep(10 * 1000);
        }
    }
}

File: dotnet/examples/election/ElectionMain.cs
namespace ElectionExample
{
  using System;
  using System.Threading;
  using System.Threading.Tasks;

  using Metaparticle.Sync;

  public class ElectionMain {
    public static void Main(string[] args) {
      var election = new Election(
        "test",
        () => {
          Console.WriteLine("I am the leader!");
        },
        () => {
          Console.WriteLine("I lost the leader!");
        });
      Console.WriteLine("Waiting for election");
      election.Run();
    }
  }
}